src/qat/model/hardware_base.py

```diff
@@ -1,3 +1,215 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import Annotated, TypeVar, get_args
+
+import numpy as np
+from frozendict import frozendict
+from pydantic import AfterValidator, RootModel
+from pydantic_core import core_schema
+
 # This base file is used to implement classes/methods common to all hardware.
 
-QubitId = int
+
+def validate_non_negative(v: int):
```

Comment:
Minor but I'd prefer `val` to v, even if it's fairly simple functionality it still is nice to be explicit.

Reply:
I'll change it, thanks!